### DISCUSSION TITLE
Calculating signal strength for the Sleepy End device

### DISCUSSION THREAD

**Discussion by nikhil30081995** [openthread/openthread#8384] [Q&A] [answered]
Hello, 

I have a sleepy end device publishing data at interval of 100 ms to cloud via Border router. Now i would like to send signal strength information between sleepy end device and its parent device that can be device connected to Border router such as NCP/RCP device OR a FTD device in network. 

So is there any API available to calculate the Signal strength info between transmiting sleepy device and its parent ? 

I came across this api call : otPlatRadioGetRssi(thread_ot_instance_get()) but it always return 127 value which is invalid. So can any help how can i calculate strength of signal or quality of signal between sleepy end device and its parent. 

Thanks in advance.

**Comment by jwhui** [comment, marked as answer]
The [`neighbor table`](https://github.com/openthread/openthread/tree/main/src/cli#neighbor-table) CLI command and [`otThreadGetNextNeighborInfo()`](https://openthread.io/reference/group/api-thread-general#otthreadgetnextneighborinfo) API provides RSSI information for each neighbor.

**Reply by nikhil1983** [reply]
Okay so if I have five SED and one FTD in network, running neighbour table command will give Signal Strength/link quality between that device and FTD device.

**Reply by nikhil1983** [reply]
> The [`neighbor table`](https://github.com/openthread/openthread/tree/main/src/cli#neighbor-table) CLI command and [`otThreadGetNextNeighborInfo()`](https://openthread.io/reference/group/api-thread-general#otthreadgetnextneighborinfo) API provides RSSI information for each neighbor.

Can I use otThreadGetParentLastRssi to get the signal strength between SED and its parent ?

**Reply by jwhui** [reply]
Yes, that can work as well.
